unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/EditorMode/BuilderInWorldGodMode.cs

```diff
@@ -137,32 +155,89 @@ public override void Init(GameObject goToEdit, GameObject undoGo, GameObject sna
         HUDController.i.buildModeHud.OnScaleSelectedAction += ScaleMode;
     }
 
+    private void MouseDrag(int buttonId, Vector3 mousePosition, float axisX, float axisY)
+    {
+        if (buttonId != 0 ||
+            selectedEntities.Count <= 0)
+            return;
+
+        if(!isDraggingStarted)
+            StarDraggingSelectedEntities();
+
+        if (canDragSelectedEntities)
+        {
+            Vector3 currentPoint = GetFloorPointAtMouse(mousePosition);
+            Vector3 direction = currentPoint - dragStartedPoint;
+
+            editionGO.transform.position += direction;
+            dragStartedPoint = currentPoint;
+        }
+
+    }
+
     private void MouseUp(int buttonID, Vector3 position)
     {
-        if (mousePressed && buttonID == 0)
+        if (buttonID != 0)
+            return;
+
+        EndDraggingSelectedEntities();
+
+        if (isMakingSquareMultiSelection && mousePressed)
```

Comment:
```suggestion
        if (isDoingSquareMultiSelection && mousePressed)
```

`Making` refers to the act of creating something, thus I think `doing` is more appropiate here.